core/src/test/scala/examples/ConsumerExample.scala

```diff
@@ -136,3 +140,55 @@ object ConsumerToProducerWithBatchCommits2Example extends ConsumerExample {
     .mapAsync(producerSettings.parallelism)(_.commit())
 }
 
+
+// Backpressure per partition with batch commit
+object ConsumerWithPerPartitionBackpressure extends ConsumerExample {
+  val control = Consumer.committablePartitionedSource(consumerSettings.withClientId("client1"))
+    .flatMapMerge(maxPartitions, _._2)
+    .via(business)
+    .batch(max = 100, first => CommittableOffsetBatch.empty.updated(first.committableOffset)) { (batch, elem) =>
+      batch.updated(elem.committableOffset)
+    }
+    .mapAsync(1)(_.commit())
+}
+
+// Flow per partition
+object ConsumerWithIndependentFlowsPerPartition extends ConsumerExample {
+  val control = Consumer.committablePartitionedSource(consumerSettings.withClientId("client1"))
+    .map { case (topicPartition, source) =>
+      source
+        .via(business)
+        .toMat(Sink.ignore)(Keep.both)
+        .run()
+    }
+    .mapAsyncUnordered(maxPartitions)(_._2)
+}
+
+// Join flows based on automatically assigned partitions
+object ConsumerWithOtherSource extends ConsumerExample {
```

Comment:
In this case, I have to join two topics. Messages in topics have the same rule of partition assignment: if the message was fetched from partition n then the corresponding message from another topic can be found in partition n too.
To prevent manual partition assignment I want to use built-in assignment from kafka for one topic, react to the assignment and perform the manual assignment to another topic.